src/go/guestagent/pkg/tracker/apitracker.go

```diff
@@ -52,25 +44,29 @@ var (
 // and unexposing the ports on the host. This should only be used when
 // the Rancher Desktop networking is enabled and the privileged service is disabled.
 type APITracker struct {
-	forwarder      forwarder.Forwarder
-	isAdmin        bool
-	baseURL        string
-	tapInterfaceIP string
-	httpClient     http.Client
-	portStorage    *portStorage
+	context           context.Context
+	wslProxyForwarder forwarder.Forwarder
+	isAdmin           bool
+	enableListeners   bool
+	baseURL           string
+	tapInterfaceIP    string
+	portStorage       *portStorage
+	apiForwarder      *forwarder.APIForwarder
 	*ListenerTracker
 }
 
 // NewAPITracker creates a new instance of a API Tracker.
-func NewAPITracker(forwarder forwarder.Forwarder, baseURL, tapIfaceIP string, isAdmin bool) *APITracker {
+func NewAPITracker(ctx context.Context, wslProxyForwarder forwarder.Forwarder, baseURL, tapIfaceIP string, isAdmin, enableListeners bool) *APITracker {
```

Comment:
There's so many args here that it needs to be documented. When is `enableListeners` relevant?

We may want to consider e.g. [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) instead.

Reply:
I added documentation for the args. Also, I like the use of functional options, however, I can't see how it could benefit us in this case.